workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/EntityGameObjectLinker.cs

```diff
@@ -43,6 +45,30 @@ public EntityGameObjectLinker(World world, ILogDispatcher logDispatcher)
             spatialOSComponent.SpatialEntityId = spatialEntityId;
             spatialOSComponent.World = world;
             spatialOSComponent.LogDispatcher = logDispatcher;
+
+            var gameObjectReference = new GameObjectReference { GameObject = gameObject };
+            viewCommandBuffer.AddComponent(entity, gameObjectReference);
+
+            var gameObjectReferenceHandleComponent = new GameObjectReferenceHandle();
+            entityCommandBuffer.AddComponent(entity, gameObjectReferenceHandleComponent);
+
+            gameObjectDispatcherSystem.CreateActivationManagerAndReaderWriterStore(entity, gameObject);
```

Comment:
I don't find it very intuitive that we call this from a system to call into a different system.  Systems should be modular by itself and there should be no need for a system to reference adifferent one (even indirect).
It would be better to factor this out into two different systems. It looks like we try to much in one system

Reply:
I don't agree that a system should never need to reference another one, especially as there isn't a concept of system components at the moment. However I do agree that it seems like this can be factored out. It seems like that function should be owned by what ever does the linking.

Reply:
We currently need to store a mapping of entity to ActivationManager somewhere. Currently it's stored in the `GameObjectDispatcherSystem`. At the same time, no matter, where the mapping is stored, the `GameObjectDispatcherSystem` needs to access it when ticked. We could start storing the mapping in the `EntityGameObjectLinker` which is currently owned by the `EntityInitializationSystem` that that would mean that the `GameObjectDispatcherSystem` needs to reference the `EntityInitializationSystem` to get the linker. Either way, as far as I can tell, all data is eventually owned by either a system or by an entity so you'll need to reference other systems somehow at the end of the day.

Reply:
This line will be removed in my PR, so it's fine to merge it in for now